Code/src/main/java/nl/utwente/group10/ui/components/blocks/function/FunctionBlock.java

```diff
@@ -0,0 +1,248 @@
+import nl.utwente.group10.haskell.type.FuncT;
+import nl.utwente.group10.haskell.type.Type;
+import nl.utwente.group10.ui.BackendUtils;
+import nl.utwente.group10.ui.CustomUIPane;
+import nl.utwente.group10.ui.components.anchors.InputAnchor;
+import nl.utwente.group10.ui.components.anchors.OutputAnchor;
+import nl.utwente.group10.ui.components.blocks.Block;
+import nl.utwente.group10.ui.components.blocks.input.InputBlock;
+import nl.utwente.group10.ui.components.blocks.output.OutputBlock;
+import nl.utwente.group10.ui.exceptions.FunctionDefinitionException;
+import nl.utwente.group10.ui.handlers.ConnectionCreationManager;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.value.ObservableValue;
+
+/**
+ * Main building block for the visual interface, this class represents a Haskell
+ * function together with it's arguments and visual representation.
+ */
+public class FunctionBlock extends Block implements InputBlock, OutputBlock {
+    /** The OutputAnchor of this FunctionBlock. */
+    private OutputAnchor output;
+
+    /** The function name. */
+    private StringProperty name;
+
+    /** The space containing the input anchor(s). */
+    @FXML private Pane inputSpace;
+
+    /** The space containing the output anchor. */
+    @FXML private BorderPane outputSpace;
+
+    /** The space containing all the arguments of the function. */
+    private ArgumentSpace argumentSpace;
+    
+    /** The space in which to nest the FunctionBlock's inner parts. */
+    @FXML private Pane nestSpace;
+    
+    /** The space in which the information of the function is displayed. */
+    @FXML private Pane functionInfo;
+    
+    /**
+     * Method that creates a newInstance of this class along with it's visual
+     * representation
+     *
+     * @param name
+     *            The name of the function.
+     * @param type
+     *            The function's type (usually a FuncT).
+     * @param pane
+     *            The parent pane in which this FunctionBlock exists.
+     */
+    public FunctionBlock(String name, Type type, CustomUIPane pane) {
+        super(pane);
+        this.name = new SimpleStringProperty(name);
+
+        this.loadFXML("FunctionBlock");
+        Expr signature = new Ident(getName());
+
+        // Collect argument types
+        ArrayList<String> args = new ArrayList<>();
+        Type t = null;
+        try {
+            t = signature.getType(pane.getEnvInstance()).prune();
+        } catch (HaskellException e1) {
+            throw new FunctionDefinitionException();
+        }
+        int inputCount = 0;
+        while (t instanceof FuncT) {
+            FuncT ft = (FuncT) t;
+            args.add(ft.getArgs()[0].toHaskellType());
+            t = ft.getArgs()[1];
+            inputCount++;
+        }
+        
+        argumentSpace = new ArgumentSpace(this, inputCount);
+        argumentSpace.setKnotIndex(getAllInputs().size());
```

Comment:
Could be, but I do not see any guarantee. Cannot answer this question in good faith.